src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -57,6 +61,40 @@ public class SSLHelper {
     CLIENT_AUTH_MAPPING.put(ClientAuth.NONE, io.netty.handler.ssl.ClientAuth.NONE);
   }
 
+  ByteBufAllocator clientByteBufAllocator(SslChannelProvider sslChannelProvider) {
+    assert sslChannelProvider != null && sslChannelProvider.sslContextProvider() != null &&
```

Comment:
@vietj these are here to compensate to lack of tests here: these should fail in the CI even if the sys prop is not used, so we know that the assumption, while enabled, is broken